packages/mermaid/src/diagrams/git/gitGraphTypes.ts

```diff
@@ -0,0 +1,133 @@
+import type { GitGraphDiagramConfig } from '../../config.type.js';
+import type { DiagramDBBase } from '../../diagram-api/types.js';
+
+export interface CommitType {
+  NORMAL: number;
+  REVERSE: number;
+  HIGHLIGHT: number;
+  MERGE: number;
+  CHERRY_PICK: number;
+}
+
+export interface Commit {
+  id: string;
+  message: string;
+  seq: number;
+  type: number;
+  tags: string[];
+  parents: string[];
+  branch: string;
+  customType?: number;
+  customId?: boolean;
+}
+
+export interface GitGraph {
+  statements: Statement[];
+}
+
+export interface Position {
+  x: number;
+  y: number;
+}
+
+export type Statement = CommitAst | BranchAst | MergeAst | CheckoutAst | CherryPickingAst;
+
+export interface CommitAst {
+  $type: 'Commit';
+  id: string;
+  message?: string;
+  tags?: string[];
+  type?: 'NORMAL' | 'REVERSE' | 'HIGHLIGHT';
+}
+
+export interface BranchAst {
+  $type: 'Branch';
+  name: string;
+  order?: number;
+}
+
+export interface MergeAst {
+  $type: 'Merge';
+  branch: string;
+  id?: string;
+  tags?: string[];
+  type?: 'NORMAL' | 'REVERSE' | 'HIGHLIGHT';
+}
+
+export interface CheckoutAst {
+  $type: 'Checkout';
+  branch: string;
+}
+
+export interface CherryPickingAst {
+  $type: 'CherryPicking';
+  id: string;
+  tags?: string[];
+  parent: string;
+}
+
+export interface GitGraphDB extends DiagramDBBase<GitGraphDiagramConfig> {
+  commitType: CommitType;
+  setDirection: (dir: DiagramOrientation) => void;
+  setOptions: (rawOptString: string) => void;
+  getOptions: () => any;
+  commit: (msg: string, id: string, type: number, tags?: string[]) => void;
+  branch: (name: string, order?: number) => void;
+  merge: (
+    otherBranch: string,
+    customId?: string,
+    overrideType?: number,
+    customTags?: string[]
+  ) => void;
+  cherryPick: (
+    sourceId: string,
+    targetId: string,
+    tags: string[] | undefined,
+    parentCommitId: string
+  ) => void;
+  checkout: (branch: string) => void;
+  prettyPrint: () => void;
+  clear: () => void;
+  getBranchesAsObjArray: () => { name: string }[];
+  getBranches: () => Map<string, string | null>;
+  getCommits: () => Map<string, Commit>;
+  getCommitsArray: () => Commit[];
+  getCurrentBranch: () => string;
+  getDirection: () => DiagramOrientation;
+  getHead: () => Commit | null;
+}
+
```

Comment:
I add the two DB partials. The reason for this was because the renderer shouldn't be able to modify the db and the parser does not need to read the data from the db. It's more important that the parser does not change data but I thought I might as well do both of them. Let me know what you think or if this is a little over kill.